jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -86,9 +86,19 @@ def on_mouse_or_key_event(self, data):
             fmt = 'x={0:0' + str(maxsize) + '.1f} y={1:0' + str(maxsize) + '.1f}'
             self.label_mouseover.pixel = (fmt.format(x, y))
 
-            if data_has_valid_wcs(image):
+            # Hack to insert WCS for generated 2D and 3D images using FLUX cube WCS.
+            if 'Plugin' in image.meta:
+                coo_data = self.jdaviz_app.data_collection[0]
```

Comment:
this assumes that `data_collection[0]` will never be from a plugin, right?  That's probably safe now (at least with the UI preventing deleting the originally loaded data), but some day we might need more advanced logic here if that assumption breaks.  (This is more a comment to my/our future selves than actually suggesting a change now).